### PR TITLE
support opt-in rules

### LINKED ISSUE #256: enabled_rules option in .swiftlint.yml?
- It would be helpful to have an `enabled_rules` option in the `.swiftlint.yml` config file that works as a whitelist. 
- I'm using swiftlint as part of a CI process, so by using the `disabled_rules` option, each time I upgrade the swift lint binary, and there have been new rules added, I have to disable more rules (otherwise swiftlint will generate warnings / errors for rules I don't necessarily want).

### PR DESCRIPTION
**Pull request by jpsim**
Closes #256 /cc @keith 

We could rules conform to this that are either not universally accepted, or applicable (e.g. #245 & #321) and rules that have too high of a false positive rate (e.g. #201 & #206).

### CODE REVIEW

**Comment by keith** [comment]
Awesome! :+1:

**Comment by daniel-beard** [comment]
This is great! So `enabled_rules:` in the `yml` file acts as a whitelist?

File: Source/SwiftLintFramework/Models/Configuration.swift [hunk 1]
```diff
@@ -78,14 +76,17 @@ public struct Configuration: Equatable {
         }
         self.disabledRules = validDisabledRules
 
-        self.rules = rules.filter {
-            !validDisabledRules.contains($0.dynamicType.description.identifier)
+        self.rules = rules.filter { rule in
+            let id = rule.dynamicType.description.identifier
+            if validDisabledRules.contains(id) { return false }
+            return enabledRules.contains(id) || !(rule is OptInRule)
```
Comment (daniel-beard) [inline Source/SwiftLintFramework/Models/Configuration.swift]: I'd prefer not to have the `OptInRule` or possibly a different mechanism for it. It makes `enablesRules` only useful for opting in to non-default rules, rather a definitive whitelist.

Comment (jpsim) [inline Source/SwiftLintFramework/Models/Configuration.swift]: I see. That's not what I was going for, and not what I outlined in https://github.com/realm/SwiftLint/issues/256#issuecomment-163393662.

What you want is a definitive list of rules to run, ignoring `enabled_rules` and `disabled_rules`, which now reading back that comment I linked to, was the motivation for a `rules` config option.

I see value in that, but it does have to be weighed against the complexity of having a white list, a black list, and a list list :disappointed: .

Comment (daniel-beard) [inline Source/SwiftLintFramework/Models/Configuration.swift]: I get it, although I got a different meaning from @scottrhoyt 's comment which seems to imply either a white list or black list, but not both (or opt in).

**Pull request merged**